package/lib/src/beam_location.dart

```diff
@@ -179,52 +188,64 @@ class SimpleBeamLocation extends BeamLocation {
   ///
   /// If none of the routes _matches_ [state.uri], nothing will be selected
   /// and [BeamerDelegate] will declare that the location is [NotFound].
-  static Map<String, String> chooseRoutes(
-      BeamState state, Iterable<String> routes) {
-    var matched = <String, String>{};
+  static Map<dynamic, String> chooseRoutes(
+      BeamState state, Iterable<dynamic> routes) {
+    var matched = <dynamic, String>{};
     bool overrideNotFound = false;
     for (var route in routes) {
-      final uriPathSegments = List.from(state.uri.pathSegments);
-      if (uriPathSegments.length > 1 && uriPathSegments.last == '') {
-        uriPathSegments.removeLast();
-      }
-
-      final routePathSegments = Uri.parse(route).pathSegments;
-
-      if (uriPathSegments.length < routePathSegments.length) {
-        continue;
-      }
+      if (route is String) {
+        final uriPathSegments = List.from(state.uri.pathSegments);
+        if (uriPathSegments.length > 1 && uriPathSegments.last == '') {
+          uriPathSegments.removeLast();
+        }
 
-      var checksPassed = true;
-      var path = '';
-      for (int i = 0; i < routePathSegments.length; i++) {
-        path += '/${uriPathSegments[i]}';
+        final routePathSegments = Uri.parse(route).pathSegments;
 
-        if (routePathSegments[i] == '*') {
-          overrideNotFound = true;
+        if (uriPathSegments.length < routePathSegments.length) {
           continue;
         }
-        if (routePathSegments[i].startsWith(':')) {
-          continue;
-        }
-        if (routePathSegments[i] != uriPathSegments[i]) {
-          checksPassed = false;
-          break;
+
+        var checksPassed = true;
+        var path = '';
+        for (int i = 0; i < routePathSegments.length; i++) {
+          path += '/${uriPathSegments[i]}';
+
+          if (routePathSegments[i] == '*') {
+            overrideNotFound = true;
+            continue;
+          }
+          if (routePathSegments[i].startsWith(':')) {
+            continue;
+          }
+          if (routePathSegments[i] != uriPathSegments[i]) {
+            checksPassed = false;
+            break;
+          }
         }
-      }
 
-      if (checksPassed) {
-        matched[route] = Uri(
-          path: path == '' ? '/' : path,
-          queryParameters:
-              state.queryParameters.isEmpty ? null : state.queryParameters,
-        ).toString();
+        if (checksPassed) {
+          matched[route] = Uri(
+            path: path == '' ? '/' : path,
+            queryParameters:
+                state.queryParameters.isEmpty ? null : state.queryParameters,
+          ).toString();
+        }
+      } else {
+        final regexp = route as RegExp;
```

Comment:
I'm wondering should we customize the "X is not a subtype of RegExp" to provide more details? (this also relates to how well we document `pathBlueprints` and `routes`)

Reply:
I'm with you on this one I think it'll minimize debugging efforts. Should I just wrap everything in a try-catch? Or do you think it could be done cleaner?

Reply:
I think a try-catch around this line 234 is alright. Then it would be great to also output the original exception text and then append our custom exception describing that it needs to be RegExp, etc.
At this point, we should be making our custom Exception classes, but let's leave that for another PR.